src/states/categories/categoriesState.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useContext, useMemo } from "react";
+import { database } from "../../firebase";
+import { ref, get } from "@firebase/database";
+import { CategoriesContext } from "./categories.context";
+
+export const useCategoriesState = () => {
+  const {
+    allCategories,
+    setAllCategories,
+    allCategoriesLoading,
+    setAllCategoriesLoading,
+  } = useContext(CategoriesContext);
+
+  const getAllCategories = useCallback(async () => {
+    setAllCategoriesLoading(true);
+    try {
+      get(ref(database, "category-names")).then((snapshot) => {
```

Comment:
you are wrapping in a try catch
don't use the .then approach of promises
use the await method.